staging/src/k8s.io/client-go/util/cert/pem.go

```diff
@@ -106,6 +108,48 @@ func ParsePrivateKeyPEM(keyData []byte) (interface{}, error) {
 	return nil, fmt.Errorf("data does not contain a valid RSA or ECDSA private key")
 }
 
+// ParsePublicKeysPEM is a helper function for reading an array of rsa.PublicKey or ecdsa.PublicKey from a PEM-encoded byte array.
+// Reads public keys from both public and private key files.
+func ParsePublicKeysPEM(keyData []byte) ([]interface{}, error) {
+	var block *pem.Block
+	keys := []interface{}{}
+	for {
+		// read the next block
+		block, keyData = pem.Decode(keyData)
+		if block == nil {
+			break
+		}
+
+		// get PEM bytes for just this block
+		blockData := pem.EncodeToMemory(block)
+		if privateKey, err := jwt.ParseRSAPrivateKeyFromPEM(blockData); err == nil {
```

Comment:
I'd rather make the x509 parse calls in directly than add a godep on this to so many kube components... with private helpers like these, we wouldn't need anything other than the stdlib:

```go
func ParsePublicKeysPEM(keyData []byte) ([]interface{}, error) {
	var block *pem.Block
	keys := []interface{}{}
	for {
		// read the next block
		block, keyData = pem.Decode(keyData)
		if block == nil {
			break
		}

		if publicKey, err := parsePublicKey(block.Bytes); err == nil {
			keys = append(keys, publicKey)
			continue
		}
		if privateRSAKey, err := parsePrivateRSAKey(block.Bytes); err == nil {
			keys = append(keys, &privateRSAKey.PublicKey)
			continue
		}
		if privateECDSAKey, err := parsePrivateECDSAKey(block.Bytes); err == nil {
			keys = append(keys, &privateECDSAKey.PublicKey)
			continue
		}

		// tolerate non-key PEM blocks for backwards compatibility
	}

	if len(keys) == 0 {
		return nil, fmt.Errorf("data does not contain any valid RSA or ECDSA public keys")
	}
	return keys, nil
}

func parsePublicKey(data []byte) (interface{}, error) {
  ...
}

func parsePrivateRSAKey(data []byte) (*rsa.PrivateKey, error) {
  ...
}

func parsePrivateECDSAKey(data []byte) (*ecdsa.PrivateKey, error) {
  ...
}
```